File: src/main/java/io/polygloat/repository/LanguageRepository.java
package io.polygloat.repository;

import io.polygloat.model.Language;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Collection;
import java.util.Optional;
import java.util.Set;

@Repository
public interface LanguageRepository extends JpaRepository<Language, Long> {
    Optional<Language> findByAbbreviationAndRepository(String abbreviation, io.polygloat.model.Repository repository);

    Optional<Language> findByNameAndRepository(String name, io.polygloat.model.Repository repository);

    Optional<Language> findByAbbreviationAndRepositoryId(String abbreviation, Long repositoryId);

    Set<Language> findAllByRepositoryId(Long repositoryId);

    Set<Language> findAllByAbbreviationInAndRepositoryId(Collection<String> abbreviation, Long repositoryId);

    void deleteAllByRepositoryId(Long repositoryId);
}

File: src/main/java/io/tolgee/repository/LanguageRepository.kt
package io.tolgee.repository

import io.tolgee.model.Language
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface LanguageRepository : JpaRepository<Language, Long> {
    fun findByAbbreviationAndProject(abbreviation: String, project: io.tolgee.model.Project): Optional<Language>
    fun findByNameAndProject(name: String?, project: io.tolgee.model.Project): Optional<Language>
    fun findByAbbreviationAndProjectId(abbreviation: String?, projectId: Long): Optional<Language>
    fun findAllByProjectId(projectId: Long?): Set<Language>
    fun findAllByAbbreviationInAndProjectId(abbreviation: Collection<String?>?, projectId: Long?): Set<Language>
    fun deleteAllByProjectId(projectId: Long?)
}
